lib/tapioca/sorbet_ext/generic_name_patch.rb

```diff
@@ -0,0 +1,54 @@
+# typed: true
+# frozen_string_literal: true
+
+require "tapioca/sorbet_ext/name_patch"
+
+module T
+  module Generic
+    # This module intercepts calls to generic type instantiations and type variable definitions.
+    # Tapioca stores the data from those calls in a `GenericTypeRegistry` which can then be used
+    # to look up the original call details when we are trying to do code generation.
+    #
+    # We are interested in the data of the `[]`, `type_member` and `type_template` calls which
+    # are all needed to generate good generic information at runtime.
+    module TypeStoragePatch
+      def [](*types)
+        Tapioca::GenericTypeRegistry.register_type(self, types)
```

Comment:
Maybe a comment here that `[]` from upstream simply returns `self`? Alternatively: use `super`

Reply:
Yeah, good call!

Reply:
Done